worker/provisioner/provisioner_task.go

```diff
@@ -837,14 +830,18 @@ func (task *provisionerTask) machineAvailabilityZoneDistribution(machine *apipro
 	} else {
 		sort.Sort(byPopulationThenNames(task.availabilityZoneMachines))
 		for _, zoneMachines := range task.availabilityZoneMachines {
-			if !zoneMachines.FailedMachineIds.Contains(machine.Id()) {
+			if !zoneMachines.FailedMachineIds.Contains(machineId) &&
+				!zoneMachines.ExcludedMachineIds.Contains(machineId) {
 				machineZone = zoneMachines.ZoneName
-				zoneMachines.MachineIds.Add(machine.Id())
+				zoneMachines.MachineIds.Add(machineId)
 				break
 			}
 		}
 	}
-	return machineZone
+	if machineZone == "" {
+		return machineZone, errors.New("availabilty zone not found, internal failure.")
```

Comment:
errors.NotFoundf("suitable availability zone for machine %v", machineId)

no need for "internal failure"

Reply:
updated